### PR TITLE
fix(Core/Dungeon): Small corrections to Boss Novos encounter and some npcs wrong behavior

### PR DESCRIPTION
**Pull request by GMKyle**
##### CHANGES PROPOSED:
-  Small corrections to Boss Novos encounter
-  Make Risen Drakkari Bat Rider(26638) aggressive
-  Prevent that sparring npcs Drakkari Guardian(26620) & Scourge Brute (26623) kill each other

##### ISSUES ADDRESSED:
Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2223

##### TESTS PERFORMED:
- Build without errors and tested in-game. (Win10 server)

##### HOW TO TEST THE CHANGES:
- Enter in Drak'Tharon Keep
- Engage Boss Novos (Boss 2)
- Risen Drakkari Bat Rider is aggressive in the next room
- In the nearly end of instance sparring npcs won't kill or attack each other.

##### Target branch(es):
- [x] Master

 

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested.No problem.

**Comment by Stoabrogga** [comment]
@GMKyle 
Under "ISSUES ADDRESSED:" Please write "Closes #2223" in one line or else the issue would not be automatically closed after this PR is merged.

**Comment by GMKyle** [comment]
Before this PR there are fighting each other for real, players can easily ignore them and move on, also watch the video if they take damage from players they will kill each other. 

taken from issue #2223
AC Core Video https://youtu.be/fgbi8wnp5Gw
The Drakkari on top of the wall only attack when you attack them.
The Drakkari on top of the wall kill themselves after the first attack.

**Comment by Stoabrogga** [comment]
Sorry, but after applying your SQL they kill themselves without player interaction:

WoWScrnShot_090619_171136

Tested it this way:
```
.gm off
.i un all
.go -479.639 -624.505 72.6095 600
```

**Comment by GMKyle** [comment]
The faction change is for prevent that behavior precisely but now I'm confuse because I've been standing near them for a while and everything works normally.
Without PR they normally take damage form each other spell but not die until take damage from players

test

**Comment by pklloveyou** [comment]
TIM截图20190907012531

**Comment by pklloveyou** [comment]
> 抱歉，在应用SQL之后，他们会在没有玩家互动的情况下自杀：
> 
> WoWScrnShot_090619_171136
> 
> 以这种方式测试：
> 
> ```
> .gm off
> .i un all
> .go -479.639 -624.505 72.6095 600
> ```

TIM截图20190907012531

**Comment by GMKyle** [comment]
You guys are right, i will try to fix it :)

**Comment by Stoabrogga** [comment]
I implemented sparring NPCs for the Skorn area:
https://github.com/azerothcore/azerothcore-wotlk/pull/1780/files#diff-b5a49ae75f84c1e7d029d153ec1b05c3R33-R83

Perhaps this helps. Basically they trigger the attack emote on the sparring partner if they are near. Has the nice side benefit of stopping the emotes if no ones near.

**Comment by GMKyle** [comment]
@pklloveyou @Stoabrogga I made some changes, can you guys test

**Comment by Stoabrogga** [comment]
@GMKyle 
Please fix the Travis error first, there's something wrong with your SQL.

**Comment by pklloveyou** [comment]
QQ截图20190911221628
QQ截图20190911221621

**Comment by Stoabrogga** [comment]
Tested, looks good now 👍 . As a nice side benefit the Drakkari Guardians now correctly use "Hamstring" and "Shield Bash" which was prevented by the SAI for the creature GUID (if there exists a SAI for both creature template entry and creature GUID, only the one with the GUID is executed).

File: data/sql/updates/pending_db_world/rev_1567713200326299359.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+DELETE FROM `creature_addon` WHERE `guid` = 127438;
+INSERT INTO `creature_addon` (`guid`, `bytes2`, `emote`) VALUES 
+(127438, 1, 36);
+
+DELETE FROM `smart_scripts` WHERE `entryorguid`= 26623  AND `id` IN (10,11);
+
+DELETE FROM `smart_scripts` WHERE `entryorguid` IN (-127591,-127590,-127589,-127582,-127580,-127579,-127578);
+
+DELETE FROM `creature_formations` WHERE `leaderGUID` IN (127590,27580,127449,127579);
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1567713200326299359.sql]:
```sql
DELETE FROM `creature_formations` WHERE `leaderGUID` IN (127590,127580,127449,127579);
```
27580->127580

Comment (Stoabrogga) [reply data/sql/updates/pending_db_world/rev_1567713200326299359.sql]: Good catch, thank you! I missed this while reviewing the PR. I have corrected the SQL script, so this is still ready to be merged.

**Pull request merged**